### DISCUSSION TITLE
Can’t verify phone number of existing user

### DISCUSSION THREAD

**Discussion by gabriel-aleixo** [supabase/supabase#7376] [Questions]
Hello! I posted this on Discord yesterday and thought I'd ask here as well.

I'm having trouble with the
`supabase.auth.update()` method in the Javascript library.
My goal is to add a verified phone number to an existing user, who authenticated initially via Magic Link.
What's boggling me is that I can make it work with a basic clean test project, but I can't get it to work with my main project!...

In order to debug the flow, I created a very simple html page, where I included the javascript library via CDN:
```
<script src="https://cdn.jsdelivr.net/npm/@supabase/supabase-js"></script>
<script>
        const supabaseUrl = 'https://{{my-project}}.supabase.co'
        const supabaseKey = '{{my-anon-key}}'
        const supa = supabase.createClient(supabaseUrl, supabaseKey)
    </script>
```
With that, I'm using the browser console directly:
`supa.auth.signIn({ email: 'some-test@emaill.com'})`
The promise is fulfilled successfully and I receive the email with the Magic Link. Once I open the Magic Link, I'm successfully authenticated.
`supa.auth.user()` shows me the user details.

Then I run:
`supa.auth.update({ phone: '+1{{my-phone-here}}' })`
Promise fulfilled successfully and I receive the code via SMS, through my Twilio account which I connected via the dashboard.

Then:
`supa.auth.verifyOTP({ phone: '+1{{my-phone-here}}', token:'XXXXXX', type: 'phone_change' })`

This promise returns an error:
```
error:
message: "Token has expired or is invalid"
status: 401
```
The Auth log on the dashboard says this:
```
time="2022-06-22T04:54:21Z" level=info msg="401: Token has expired or is invalid" component=api error="redirect user" method=POST path=/verify referer="http://localhost:3000/" remote_addr="98.59.249.2:30283" request_id=97983326-7a91-44bb-990b-51e0052b16a5
```
The crazy part is that these exact same steps work fine with another project, and I get the phone number added to the user. I can't find any difference in the settings of these two projects.

Does anybody have any idea what I'm doing wrong? Thanks!

**Comment by kangmingtay** [comment]
hi @gabriel-aleixo, please reach out to us at https://supabase.com/support so we can take a look and see if it's specific to your project!

**Comment by maxhvlm** [comment]
I'm having the exact same issue, were you able to resolve this after contacting support?

**Comment by omtodkar** [comment]
Same here, did support resolved your issue? Facing issue on local environment. Not even working for test the phone numbers.

**Console Log**
```log
{
  __isAuthError: true,
  status: 403,
  code: 'otp_expired'
}
AuthApiError: Token has expired or is invalid.
```

**API Gateway Log**
```json
{
  "event_message": "172.19.0.1 - - [20/Nov/2024:17:27:07 +0000] \"POST /auth/v1/verify HTTP/1.1\" 403 66 \"-\" \"node\"",
  "id": "a025050f-c6fb-4b48-ac3d-c2632075cb04",
  "metadata": [
    {
      "request": [
        {
          "headers": [
            {
              "cf_connecting_ip": "172.19.0.1",
              "user_agent": "node"
            }
          ],
          "method": "POST",
          "path": "/auth/v1/verify",
          "protocol": "HTTP/1.1"
        }
      ],
      "response": [
        {
          "status_code": 403
        }
      ]
    }
  ],
  "timestamp": 1732123627000000
}
```

**Typescript API Call**
```typescript
const { data, error } = await supabase.auth.verifyOtp({
    phone: parsedInput?.phoneNumber,
    token: parsedInput?.otp,
    type: 'phone_change',
});
```

**config.toml**
```toml
[auth.email]
enable_signup = true
enable_confirmations = true

[auth.sms]
enable_signup = true
enable_confirmations = true

[auth.sms.test_otp]
919999999999 = "123456"
```
